fido/src/main/java/com/yubico/yubikit/fido/webauthn/PublicKeyCredentialCreationOptions.java

```diff
@@ -112,11 +112,18 @@ public Extensions getExtensions() {
         return extensions;
     }
 
-    public Map<String, ?> toMap() {
+    public Map<String, ?> toMap(SerializationType serializationType) {
         Map<String, Object> map = new HashMap<>();
         map.put(RP, rp.toMap());
-        map.put(USER, user.toMap());
-        map.put(CHALLENGE, Base64.encode(challenge));
+        map.put(USER, user.toMap(serializationType));
+        switch (serializationType) {
+            case JSON:
+                map.put(CHALLENGE, Base64.encode(challenge));
+                break;
+            case CBOR:
+                map.put(CHALLENGE, challenge);
+                break;
+        }
         List<Map<String, ?>> paramsList = new ArrayList<>();
         for (PublicKeyCredentialParameters params : pubKeyCredParams) {
             paramsList.add(params.toMap());
```

Comment:
Are we missing serializationType here?

Reply:
PublicKeyCredentialParameters does not have any byte[] so there is no serialization difference between JSON and CBOR so no parameter was added to toMap/fromMap methods

Reply:
Added the parameter to all webauthn types